src/google/cloud/ndb/model.py

```diff
@@ -4812,13 +4815,20 @@ def _put_async(
 
         @tasklets.tasklet
         def put(self):
+            context = context_module.get_context()
+            use_remote_cache = context._use_remote_cache(self.key, _options)
+            if use_remote_cache and self._has_complete_key():
+                yield remote_cache.cache_set_locked(self.key)
+
             entity_pb = _entity_to_protobuf(self)
             key_pb = yield _datastore_api.put(entity_pb, _options)
             if key_pb:
                 ds_key = helpers.key_from_protobuf(key_pb)
                 self._key = key_module.Key._from_ds_key(ds_key)
 
-                context = context_module.get_context()
+                if use_remote_cache and not context.in_transaction():
+                    yield remote_cache.cache_delete(self.key)
```

Comment:
Any reason not to go ahead and store the entity here?

Reply:
This is just a copy from py27 implementation.

https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/context.py#L684